C/services/storage/storage_registry.cpp

```diff
@@ -209,11 +221,72 @@ StorageRegistry::sendPayload(const string& url, char *payload)
 void
 StorageRegistry::filterPayload(const string& url, char *payload, const string& asset)
 {
+ostringstream convert;
+
 	size_t found = url.find_first_of("://");
 	size_t found1 = url.find_first_of("/", found + 3);
 	string hostport = url.substr(found+3, found1 - found - 4);
 	string resource = url.substr(found1);
 
+	// Filter the payload to include just the one asset
+	Document doc;
+	doc.Parse(payload);
+	if (doc.HasParseError())
+	{
+		Logger::getLogger()->error("filterPayload: Parse error in payload");
+		return;
+	}
+	if (!doc.HasMember("readings"))
+	{
+		Logger::getLogger()->error("filterPayload: payload has no readings object");
+		return;
+	}
+	const Value& readings = doc["readings"];
+	if (!readings.IsArray())
+	{
+		Logger::getLogger()->error("filterPayload: payload readings object is not an array");
+		return;
+	}
+	convert << "{ \"readings\" : [ ";
+	int count = 0;
+	/*
+	 * Loop over the readings and create a reading object for
+	 * each, check if it matches the asset name and incldue in the
+	 * new payload if it does. In eother case free that object
+	 * immediately to reduce the memory requirement.
+	 */
+	for (auto& reading : readings.GetArray())
+	{
+		if (reading.IsObject())
+		{
+			JSONReading *value = new JSONReading(reading);
+			if (value->getAssetName().compare(asset) == 0)
+			{
+				if (count)
+					convert << ",";
+				count++;
+				convert << value->toJSON();
+			}
+			delete value;
+		}
+	}
+	
+	convert << "] }";
+
+	/*
+	 * Check if any assets inthe filtered payload
+	 */
+	if (count == 0)
+	{
+		Logger::getLogger()->error("filterPayload: nothing to send, %s, %s", asset.c_str(), payload);
+		// Nothing to send
+		return;
+	}
+
 	HttpClient client(hostport);
-	client.request("POST", resource, payload);
+	try {
+		client.request("POST", resource, convert.str());
```

Comment:
I suggest adding timeouts in seconds:

client.config.timeout = 10;
client.config.timeout_connect = 10;